raiden/messages.py

```diff
@@ -1832,6 +1832,78 @@ def verify_request_monitoring(
+    """ Represents the settlement required message for the LC when we need the LC to sign and send a settlement """
+    def __init__(self, channel_identifier: ChannelID, channel_network_identifier: TokenNetworkAddress,
+                 participant1: Address, participant1_transferred_amount: TokenAmount,
+                 participant1_locked_amount: TokenAmount, participant1_locksroot: Locksroot, participant2: Address,
+                 participant2_transferred_amount: TokenAmount, participant2_locked_amount: TokenAmount,
+                 participant2_locksroot: Locksroot, **kwargs):
+        super().__init__(**kwargs)
+        self.channel_identifier = channel_identifier
+        self.channel_network_identifier = channel_network_identifier
+        self.participant1 = participant1
+        self.participant1_transferred_amount = participant1_transferred_amount
+        self.participant1_locked_amount = participant1_locked_amount
+        self.participant1_locksroot = participant1_locksroot
+        self.participant2 = participant2
+        self.participant2_transferred_amount = participant2_transferred_amount
+        self.participant2_locked_amount = participant2_locked_amount
+        self.participant2_locksroot = participant2_locksroot
+
+    @classmethod
+    def unpack(cls, packed) -> "SettlementRequiredLightMessage":
+        return cls(channel_identifier=packed.channel_identifier,
+                   channel_network_identifier=packed.channel_network_identifier,
+                   participant1=packed.participant1,
+                   participant1_transferred_amount=packed.participant1_transferred_amount,
+                   participant1_locked_amount=packed.participant1_locked_amount,
+                   participant1_locksroot=packed.participant1_locksroot,
+                   participant2=packed.participant2,
+                   participant2_transferred_amount=packed.participant2_transferred_amount,
+                   participant2_locked_amount=packed.participant2_locked_amount,
+                   participant2_locksroot=packed.participant2_locksroot)
+
+    def pack(self, packed) -> None:
+        packed.channel_identifier = self.channel_identifier
+        packed.channel_network_identifier = self.channel_network_identifier
+        packed.participant1 = self.participant1
+        packed.participant1_transferred_amount = self.participant1_transferred_amount
+        packed.participant1_locked_amount = self.participant1_locked_amount
+        packed.participant1_locksroot = self.participant1_locksroot
+        packed.participant2 = self.participant2
+        packed.participant2_transferred_amount = self.participant2_transferred_amount
+        packed.participant2_locked_amount = self.participant2_locked_amount
+        packed.participant2_locksroot = self.participant2_locksroot
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "channel_identifier": self.channel_identifier,
```

Comment:
"type" property  is missing

Reply:
The other messages have this line in their to_dict implementation: `type": self.__class__.__name__,`. If i didn't missunderstand its meaning, that propety is the one used later by the light client to decide what kind of message it is, and what should it do in response to it.

Reply:
Well good catch there, but is not used by the Light Client, it's used by the node only to verify that the message is correctly deserialized and serialized from database. The LC uses the LightClientProtocolMessageType to handle the message. Anyway i will add that